src/dx/pubnub_client.rs

```diff
@@ -227,9 +227,8 @@ where
                 Ok(PubNubClientRef {
                     transport: PubNubMiddleware {
                         transport: pre_build.transport,
-                        // TODO: String -> Cow<'static, str>
-                        instance_id: pre_build.instance_id.clone(),
-                        user_id: pre_build.config.user_id.clone(),
+                        instance_id: Arc::new(pre_build.instance_id.clone()),
+                        user_id: Arc::new(pre_build.config.user_id.clone()),
                         signature_keys: pre_build.config.clone().signature_key_set()?,
                     },
                     instance_id: pre_build.instance_id,
```

Comment:
But wait. `instance_id` is still not a `arc`.
Both `instance_id` fields have to be in `arc` to make a sense.

Reply:
So at `PubNubClientRef` we should make`instance_id` as `Arc<Option<String>>` right?

Reply:
Same for `user_id`. 

Both `instance_id` and `user_id` should be under the Arc for sharing the resource. 
My bad that I haven't noticed that earlier.

Reply:
✅

Reply:
@Xavrax 
I think `Arc::clone(&pre_build.instance_id)` looks more appropriate here. I've updated it but somehow I don't know why github ui is not showing it previous change outdated .